test/unit/models/consent.test.ts

```diff
@@ -0,0 +1,174 @@
+  credentialStatus: 'PENDING',
+  credentialChallenge: 'xyhdushsoa82w92mzs',
+  credentialPayload: 'dwuduwd&e2idjoj0w'
+}
+
+/*
+ * Consent Resource Model Unit Tests
+ */
+describe('consent', (): void => {
+  let Db: Knex
+  let consentModel: ConsentModel
+
+  beforeAll(async (): Promise<void> => {
+    Db = Knex(Config.test)
+    await Db.migrate.latest()
+
+    consentModel = new ConsentModel(Db)
+  })
+
+  afterAll(async (): Promise<void> => {
+    Db.destroy()
+  })
+
+  describe('registerConsent', (): void => {
+    // Reset table for new test
+    beforeEach(async (): Promise<void> => {
+      await Db<Consent>('Consent').del()
+    })
+
+    it('adds consent with partial info to the database', async (): Promise<void> => {
+      // Action
+      await consentModel.registerConsent(partialConsent)
+
+      // Assertion
+      const users: Consent[] = await Db<Consent>('Consent')
+        .select('*')
+        .where({
+          id: partialConsent.id
+        })
+
+      delete users[0].timeStamp
+
+      expect(users[0]).toEqual({
+        id: '1234',
+        initiatorId: 'pisp-2342-2233',
+        participantId: 'dfsp-3333-2123',
+        credentialId: null,
+        credentialType: null,
+        credentialStatus: null,
+        credentialPayload: null,
+        credentialChallenge: null
+      })
+    })
+  })
+
+  describe('updateCredentialsByConsentId', (): void => {
+    // Reset table for new test
+    beforeEach(async (): Promise<void> => {
+      await Db<Consent>('Consent').del()
+      await Db<Consent>('Consent')
+        .insert(partialConsent)
+    })
+
+    it('updates credentials for existing consent', async (): Promise<void> => {
+      // Action
+      await consentModel.updateCredentialsByConsentId(completeConsent)
+
+      // Assertion
+      const users: Consent[] = await Db<Consent>('Consent')
+        .select('*')
+        .where({
+          id: completeConsent.id
+        })
+
+      delete users[0].timeStamp
+
+      expect(users[0]).toEqual(completeConsent)
+    })
+  })
+
+  describe('getConsentById', (): void => {
+    // Reset table for new test
+    beforeEach(async (): Promise<void> => {
+      await Db<Consent>('Consent').del()
+      await Db<Consent>('Consent')
+        .insert(completeConsent)
+    })
+
+    it('retrieves an existing consent', async (): Promise<void> => {
+      // Action
+      const users: Consent[] = await consentModel.getConsentById(completeConsent.id)
+
+      delete users[0].timeStamp
+
+      // Assertion
+      expect(users[0]).toEqual(completeConsent)
+    })
+  })
+
+  describe('deleteConsentById', (): void => {
+    // Reset table for new test
+    beforeEach(async (): Promise<void> => {
+      await Db<Consent>('Consent').del()
+      await Db<Consent>('Consent')
+        .insert(completeConsent)
+    })
+
+    it('deletes an existing consent', async (): Promise<void> => {
+      // Pre action Assertion
+      let users: Consent[] = await Db('Consent')
```

Comment:
- The variable name `users` don't fit well. Maybe `consents` or `userConsents` would fit better?
- Use `const` instead of `let` if you are not reassigning the value.

Reply:
Addressed in a recent commit